### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by joaopamaral**
This initial release introduces the `OrganizationStream`, `TimeReportStream`, and `ContractTimeReportStream` which were developed based on UpWork new GraphQL API documentation. https://www.upwork.com/developer/documentation/graphql/api/docs/index.html

> The base project was generated using the meltano singer SDK (https://sdk.meltano.com/en/latest/) with all recommended practices for this type of extractor.

## Tests

### Tap-Upwork

```
tap-upwork --config config.json --discover > ./catalog.json
tap-upwork --config config.json --catalog catalog.json
```
### Meltano

```
meltano elt tap-upwork target-jsonl
```

### Unittest

```
tests/unittests/test_client.py::TestClient::test_property_list_to_graphql_query PASSED
```

### Core tests

```
collected 46 items                                                                                                                                                                                                                                                                                                                                                                                                         

tests/test_core.py::TestTapUpWork::test_tap_cli_prints PASSED                                                                                                                                                                                                                                                                                                                                                        [  2%]
tests/test_core.py::TestTapUpWork::test_tap_discovery PASSED                                                                                                                                                                                                                                                                                                                                                         [  4%]
tests/test_core.py::TestTapUpWork::test_tap_stream_connections PASSED                                                                                                                                                                                                                                                                                                                                                [  6%]
tests/test_core.py::TestTapUpWork::test_tap_stream_catalog_schema_matches_record[contractTimeReport] PASSED                                                                                                                                                                                                                                                                                                          [  8%]
tests/test_core.py::TestTapUpWork::test_tap_stream_catalog_schema_matches_record[organization] PASSED                                                                                                                                                                                                                                                                                                                [ 10%]
tests/test_core.py::TestTapUpWork::test_tap_stream_catalog_schema_matches_record[timeReport] PASSED                                                                                                                                                                                                                                                                                                                  [ 13%]
tests/test_core.py::TestTapUpWork::test_tap_stream_record_matches_stream_schema[contractTimeReport] PASSED                                                                                                                                                                                                                                                                                                           [ 15%]
tests/test_core.py::TestTapUpWork::test_tap_stream_record_matches_stream_schema[organization] PASSED                                                                                                                                                                                                                                                                                                                 [ 17%]
tests/test_core.py::TestTapUpWork::test_tap_stream_record_matches_stream_schema[timeReport] PASSED                                                                                                                                                                                                                                                                                                                   [ 19%]
tests/test_core.py::TestTapUpWork::test_tap_stream_record_schema_matches_catalog[contractTimeReport] PASSED                                                                                                                                                                                                                                                                                                          [ 21%]
tests/test_core.py::TestTapUpWork::test_tap_stream_record_schema_matches_catalog[organization] PASSED                                                                                                                                                                                                                                                                                                                [ 23%]
tests/test_core.py::TestTapUpWork::test_tap_stream_record_schema_matches_catalog[timeReport] PASSED                                                                                                                                                                                                                                                                                                                  [ 26%]
tests/test_core.py::TestTapUpWork::test_tap_stream_returns_record[contractTimeReport] PASSED                                                                                                                                                                                                                                                                                                                         [ 28%]
tests/test_core.py::TestTapUpWork::test_tap_stream_returns_record[organization] PASSED                                                                                                                                                                                                                                                                                                                               [ 30%]
tests/test_core.py::TestTapUpWork::test_tap_stream_returns_record[timeReport] PASSED                                                                                                                                                                                                                                                                                                                                 [ 32%]
tests/test_core.py::TestTapUpWork::test_tap_stream_primary_keys[contractTimeReport] PASSED                                                                                                                                                                                                                                                                                                                           [ 34%]
tests/test_core.py::TestTapUpWork::test_tap_stream_primary_keys[organization] PASSED                                                                                                                                                                                                                                                                                                                                 [ 36%]
tests/test_core.py::TestTapUpWork::test_tap_stream_primary_keys[timeReport] PASSED                                                                                                                                                                                                                                                                                                                                   [ 39%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_boolean[organization.active] PASSED                                                                                                                                                                                                                                                                                                                  [ 41%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_boolean[organization.hidden] PASSED                                                                                                                                                                                                                                                                                                                  [ 43%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_datetime[contractTimeReport.dateWorkedOn] PASSED                                                                                                                                                                                                                                                                                                     [ 45%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_datetime[contractTimeReport.weekWorkedOn] PASSED                                                                                                                                                                                                                                                                                                     [ 47%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_datetime[contractTimeReport.monthWorkedOn] PASSED                                                                                                                                                                                                                                                                                                    [ 50%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_datetime[contractTimeReport.yearWorkedOn] PASSED                                                                                                                                                                                                                                                                                                     [ 52%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_datetime[timeReport.dateWorkedOn] PASSED                                                                                                                                                                                                                                                                                                             [ 54%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_datetime[timeReport.weekWorkedOn] PASSED                                                                                                                                                                                                                                                                                                             [ 56%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_datetime[timeReport.monthWorkedOn] PASSED                                                                                                                                                                                                                                                                                                            [ 58%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_datetime[timeReport.yearWorkedOn] PASSED                                                                                                                                                                                                                                                                                                             [ 60%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_numeric[contractTimeReport.totalHoursWorked] PASSED                                                                                                                                                                                                                                                                                                  [ 63%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_numeric[contractTimeReport.totalCharges] PASSED                                                                                                                                                                                                                                                                                                      [ 65%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_numeric[contractTimeReport.totalOnlineHoursWorked] PASSED                                                                                                                                                                                                                                                                                            [ 67%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_numeric[contractTimeReport.totalOnlineCharge] PASSED                                                                                                                                                                                                                                                                                                 [ 69%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_numeric[contractTimeReport.totalOfflineHoursWorked] PASSED                                                                                                                                                                                                                                                                                           [ 71%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_numeric[contractTimeReport.totalOfflineCharge] PASSED                                                                                                                                                                                                                                                                                                [ 73%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_numeric[timeReport.totalHoursWorked] PASSED                                                                                                                                                                                                                                                                                                          [ 76%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_numeric[timeReport.totalCharges] PASSED                                                                                                                                                                                                                                                                                                              [ 78%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_numeric[timeReport.totalOnlineHoursWorked] PASSED                                                                                                                                                                                                                                                                                                    [ 80%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_numeric[timeReport.totalOnlineCharge] PASSED                                                                                                                                                                                                                                                                                                         [ 82%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_numeric[timeReport.totalOfflineHoursWorked] PASSED                                                                                                                                                                                                                                                                                                   [ 84%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_numeric[timeReport.totalOfflineCharge] PASSED                                                                                                                                                                                                                                                                                                        [ 86%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_object[contractTimeReport.freelancer] PASSED                                                                                                                                                                                                                                                                                                         [ 89%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_object[contractTimeReport.team] PASSED                                                                                                                                                                                                                                                                                                               [ 91%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_object[organization.flag] PASSED                                                                                                                                                                                                                                                                                                                     [ 93%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_object[timeReport.freelancer] PASSED                                                                                                                                                                                                                                                                                                                 [ 95%]
tests/test_core.py::TestTapUpWork::test_tap_stream_attribute_is_object[timeReport.team] PASSED                                                                                                                                                                                                                                                                                                                       [ 97%]
tests/unittests/test_client.py::TestClient::test_property_list_to_graphql_query PASSED 
```

> The core tests provided by singer sdk were removed from the CI tests because it required a valid config file (with credentials)

### CODE REVIEW

File: tests/unittests/test_client.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    StringType,
+    IntegerType,
+    NumberType,
+    ObjectType,
+)
+
+from tap_upwork.client import UpWorkStream
+
+
+class TestClient(TestCase):
```
Comment (joaopamaral) [inline tests/unittests/test_client.py:15]: All other tests run from the SDK (core tests).

**Review by Khrol — COMMENTED** [review]
@jessie-ross  it's an interesting case for the Python code that we can't put into our monorepo.

@joaopamaral why do we use `poetry`, BTW? Maybe `pipenv` would be easier to maintain if all our internal infra is with `pipenv`?

**Comment by jessie-ross** [comment]
If we are releasing public code then it must be separate but that is fine as we won't be using any internal packages and will be releasing it to pypi instead of our internal Nexus repo.

This means we can take advantage of poetry - I would recommend it over pipenv - poetry is the best choice if we aren't doing anything custom, it is nice and opinionated and seems to do everything out of the box.

These separate public packages can seamlessly be required and used internally within the monorepo.

**Comment by joaopamaral** [comment]
> why do we use `poetry`, BTW? Maybe `pipenv` would be easier to maintain if all our internal infra is with `pipenv`?

The `poetry` was already included by the SDK (when it generates the base project that is in the trunk), and it's recommended in the Meltano guides (https://sdk.meltano.com/en/latest/guides/porting.html#install-dependencies).

**Review by Khrol — APPROVED** [review]

**Pull request merged**